### ISSUE TITLE
Duplicates multi-word tags (etc.) when using string values and SELECT element

### ISSUE THREAD

**Issue by jimallman** [bootstrap-tagsinput/bootstrap-tagsinput#294]
**TL;DR: Any tag string that is modified by JS `escape()` is prone to unwanted duplication!**

First, thanks for this plugin! It's been hassle-free for us until just today. As described in [this GitHub issue](https://github.com/OpenTreeOfLife/opentree/issues/562#issuecomment-72159441), I've found a bug that results in creating duplicate tags when using bootstrap-tagsinput with a `select` widget. If the `select` is pre-populated with tags in its `option`s (which means we're using simple strings as our values), any multi-word tag &mdash; actually, any tag that is modified by `escape()` or `encodeURIComponent()` &mdash; will be duplicated each time the TagsInput behavior is applied.

In this setup, the tags widget reads `option` values from a chosen `select` widget, and scans its own `TagsInput` object for matching values. But for simple string values, it messes things up:

When [it first scans their `value` attributes](https://github.com/TimSchlechter/bootstrap-tagsinput/blob/0.3.9/src/bootstrap-tagsinput.js#L362), the values are read "directly" without transformation. So for this `option`, the value is read as "delete me":

``` html
    <option value="delete me">delete me</option>
```

But when it looks for matching `option` elements (to decide whether a new one needs to be created), [it checks for an URL-encoded version of this value](https://github.com/TimSchlechter/bootstrap-tagsinput/blob/0.3.9/src/bootstrap-tagsinput.js#L112). This would only match on an `option` like:

``` html
    <option value="delete%20me">delete me</option>
```

If we try to pre-escape the `value` attributes, we'll start with `option`s like the one above. But then these are scanned as initial tag values, and the test ends up looking for this:

``` html
    <option value="delete%2520me">delete me</option>
```

As far as I can see, this wil **always** fail with a multi-word tag (any string modified by `encodeURIComponent`). 

For now, I've [worked around the problem](https://github.com/OpenTreeOfLife/opentree/commit/d6f53a7f78121566f4b2fe7cd6c2ddec4687a417) by switching from simple string tag values to simple objects like

``` json
    {"displayText": "delete me", "attrValue": "delete%20me" }
```

... then specifying these properties in my tags `options`:

``` javascript
    itemValue: 'attrValue',
    itemText: 'displayString',
```

I checked the latest code, and it seems [both conflicting operations](https://github.com/TimSchlechter/bootstrap-tagsinput/blob/a4657365808a224fe8aa950c14b96ecfce58b290/src/bootstrap-tagsinput.js#L135) are [still unchanged](https://github.com/TimSchlechter/bootstrap-tagsinput/blob/a4657365808a224fe8aa950c14b96ecfce58b290/src/bootstrap-tagsinput.js#L446). If the Right Thing to do is to URI-encode attribute values, then [this should be done here](https://github.com/TimSchlechter/bootstrap-tagsinput/blob/a4657365808a224fe8aa950c14b96ecfce58b290/src/bootstrap-tagsinput.js#L135) and also specified for any intial `option` elements in the page. And the initial reading of `option` elements into tag values should un-escape these strings before adding them as tags:

``` javascript
  var tagValue = decodeURIComponent( $(this).attr('value') );
  self.add(tagValue, true);
```

If needed, I can put together a proper pull request for this, but I won't have time for a week or so.

**Comment by stickbyatlas** [comment]
I just noticed the same behaviour described above, and discovered this issue when I was browsing to see if anyone else has reported it. Thanks for posting such a detailed description. Did you find a way to continue using simple string values with multi-word tags? Surprised nobody else has been bitten by this.

**Comment by Aeramor** [comment]
I have also been bitten by this bug, Only noticed it yesterday.
Thanks for the detailed post, if you haven't written a pull request already I will.

**Comment by Hoverbear** [comment]
+1 this is still an issue.

**Comment by Dr4K4n** [comment]
The pull request has been merged and the problem is fixed, but please note that the current release (0.5.0) does not contain this fix.

**Comment by jimallman** [comment]
@Dr4K4n, can you link to the merged pull request? I'm curious to see whether it handles quotes, angle brackets, etc. safely.

**Comment by Dr4K4n** [comment]
@jimallman Well, sry, I must have looked in the wrong places... It doesn't seem to be merged yet.

**Comment by 1bnr** [comment]
this is still a problem, and it also applies to any tag with special characters that would be escaped by encodeURI. The problem is that the check for existing options checks with the value escaped with encodeURI, so of course it will not match (not sure exactly how this results in a double entry, but it does); my fix was to gather the current options first, then check if the value is among them, since removing the encodeURI from the selector causes errors in the case of values containing double quotes. at line 140 change this:
`if (self.isSelect && !$('option[value="' + encodeURIComponent(itemValue) + '"]',self.$element)[0]) {`
to this:
`      var opts = self.$element.find('option').map(function() {return $(this).val();}).get();
      if (self.isSelect && !opts.includes(itemValue)) {`
